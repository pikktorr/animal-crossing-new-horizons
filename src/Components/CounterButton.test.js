import React from "react";
import CounterButton from "./CounterButton";
import { shallow } from "enzyme";

describe("Testing stateful component", () => {
  test("expect to render CounterButton ", () => {
    const mockColor = "green";
    const wrapper = shallow(<CounterButton color={mockColor} />);

    expect(wrapper.debug()).toMatchSnapshot();
  });

  test("expect to simulate click", () => {
    const mockColor = "green";
    const wrapper = shallow(<CounterButton color={mockColor} />);

    wrapper.find('[id="button"]').simulate("click");
    wrapper.find('[id="button"]').simulate("click");
    wrapper.find('[id="button"]').simulate("keypress");
    
    expect(wrapper.state()).toEqual({ count: 2 });
    expect(wrapper.props().color).toEqual("green");
  });
});
