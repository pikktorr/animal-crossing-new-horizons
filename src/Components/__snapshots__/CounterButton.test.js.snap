// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing stateful component expect to render CounterButton  1`] = `
"<button id=\\"button\\" color=\\"green\\" onClick={[Function: onClick]}>
  Count: 
  0
</button>"
`;
